src/otx/v2/adapters/torch/mmengine/mmdet/modules/datasets/dataset.py

```diff
@@ -0,0 +1,154 @@
+"""Base MMDataset for Detection Task."""
+
+# Copyright (C) 2022 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+
+from __future__ import annotations
+
+from typing import Any, Sequence
+
+import numpy as np
+from mmcv.transforms import Compose
+from mmdet.datasets import BaseDetDataset
+from mmdet.registry import DATASETS
+
+from otx.v2.api.entities.datasets import DatasetEntity
+from otx.v2.api.entities.label import LabelEntity
+from otx.v2.api.entities.utils.data_utils import get_old_new_img_indices
+
+
+@DATASETS.register_module()
+class OTXDetDataset(BaseDetDataset):
+    """Wrapper that allows using a OTX dataset to train mmdetection models.
+
+    This wrapper is not based on the filesystem,
+    but instead loads the items here directly from the OTX DatasetEntity object.
+
+    The wrapper overwrites some methods of the BaseDetDataset class: prepare_train_img, prepare_test_img and prepipeline
+    Naming of certain attributes might seem a bit peculiar but this is due to the conventions set in BaseDetDataset. For
+    instance, CustomDatasets expects the dataset items to be stored in the attribute data_infos, which is why it is
+    named like that and not dataset_items.
+
+    """
+
+    class _DataInfoProxy:
```

Comment:
I think we can unifty OTXDataset for each tasks(det, cls, seg, action) since MM library unify their dataset api by BaseDataset in MMEngine. Let's discuss about how we can unify OTXDataset across task after phase 1